sklearn/model_selection/tests/test_validation.py

```diff
@@ -914,6 +914,10 @@ def test_cross_val_predict_with_method():
     X, y = shuffle(X, y, random_state=0)
     classes = len(set(y))
 
+    y[:50] = 0
```

Comment:
It would be more normal to make a separate test for tricky cases, rather than overload the test for basic functionality. This helps diagnose, when a test fails, what went wrong.

Certainly, this dataset modification needs a comment as to its purpose.